dandi/tests/test_upload.py

```diff
@@ -212,6 +212,19 @@ def test_upload_bids_validation_ignore(
     dandiset.get_asset_by_path("sub-Sub1/anat/sub-Sub1_T1w.nii.gz")
 
 
+def test_upload_bids_metadata(
+    mocker: MockerFixture, bids_dandiset: SampleDandiset
+) -> None:
+    bids_dandiset.upload(existing="force")
+    dandiset = bids_dandiset.dandiset
+    # Automatically check all files, heuristic should remain very BIDS-stable
+    for asset in dandiset.get_assets(order="path"):
+        apath = asset.path
+        if "Sub1" in apath:
```

Comment:
Just test on some known paths since if we later rename from Sib1, this test would pass without testing anything

Reply:
How about 288715e6d47a3af54ecb140600f5c8f880180988 ? I like that we're checking all of them, at trivial additional time cost.

Reply:
it is not runtime which concerned me but possible absent checking if that dataset changes so that eg there is no subjects -- this test would still be valid. ok  let's consider it unlikely and proceed